tokio/src/runtime/thread_pool/idle.rs

```diff
@@ -89,22 +89,15 @@ impl Idle {
     }
 
     pub(super) fn transition_worker_to_searching(&self) -> bool {
-        // Using `Relaxed` ordering is acceptable here as it is just an
-        // optimization. This load has does not need to synchronize with
-        // anything, and the algorithm is correct no matter what the load
-        // returns (as in, it could return absolutely any `usize` value and the
-        // pool would be correct.
-        let state = State::load(&self.state, Relaxed);
+        let state = State::load(&self.state, SeqCst);
         if 2 * state.num_searching() >= self.num_workers {
             return false;
         }
 
         // It is possible for this routine to allow more than 50% of the workers
         // to search. That is OK. Limiting searchers is only an optimization to
         // prevent too much contention.
-        //
-        // At this point, we do not need a hard synchronization with `notify_work`, so `AcqRel` is sufficient.
-        State::inc_num_searching(&self.state, AcqRel);
+        State::inc_num_searching(&self.state, SeqCst);
```

Comment:
Is changing these to `SeqCst` also part of the fix?

Reply:
Kind of... It's probably best to stick exclusively w/ SeqCst.